doc/developers/contributing.rst

```diff
@@ -268,100 +249,115 @@ If any of the above seems like magic to you, then look up the `Git documentation
+     examples are correct
+   - `pytest sklearn/linear_model/tests/test_logistic.py` to run the tests
+     specific to the file
+   - `pytest sklearn/linear_model` to test the whole `linear_model` module
+   - `pytest sklearn/doc/linear_model.rst` to make sure the user guide
+     examples are correct.
+
+   There may be other failing tests, but they will be caught by the CI so
+   you don't need to run the whole test suite locally. Read more in
+   `testing_coverage`_.
+
+2. **Make sure your code is properly commented and documented**, and **make
+   sure the documentation renders properly**. To build the documentation, please
+   refer to our `contribute_documentation`_ guidelines.
+
+3. **Tests are necessary for enhancements to be
+   accepted**. Bug-fixes or new features should be provided with
+   `non-regression tests
+   <https://en.wikipedia.org/wiki/Non-regression_testing>`_. These tests
+   verify the correct behavior of the fix or feature. In this manner, further
+   modifications on the code base are granted to be consistent with the
+   desired behavior. For the case of bug fixes, at the time of the PR, the
+   non-regression tests should fail for the code base in the master branch
+   and pass for the PR code.
+
+4. **Make sure that your PR does not add PEP8 violations**. Please run
+   `./build_tools/circle/flake8_diff.sh` or `make flake8-diff` on a Unix-like
+   system. You can also directly use `flake8 path_to_file`.
+
+5. Follow the `coding-guidelines`_ (see below).
+
+6. When applicable, use the validation tools and scripts in the
+   ``sklearn.utils`` submodule.  A list of utility routines available
+   for developers can be found in the :ref:`developers-utils` page.
+
+7. Give your pull request a helpful title that summarises what your
+   contribution does. In some cases "Fix <ISSUE TITLE>" is enough.
+   "Fix #<ISSUE NUMBER>" is not enough.
+
+8. Often pull requests resolve one or more other issues (or pull requests).
+   If merging your pull request means that some other issues/PRs should
+   be closed, you should `use keywords to create link to them
+   <https://github.com/blog/1506-closing-issues-via-pull-requests/>`_
+   (e.g., ``Fixes #1234``; multiple issues/PRs are allowed as long as each
+   one is preceded by a keyword). Upon merging, those issues/PRs will
+   automatically be closed by GitHub. If your pull request is simply
+   related to some other issues/PRs, create a link to them without using
+   the keywords (e.g., ``See also #1234``).
+
+9. Please prefix the title of your pull request with ``[MRG]`` if the
+   contribution is complete and should be subjected to a detailed review.
+   Two core developers will review your code and change the prefix of the pull
+   request to ``[MRG + 1]`` and ``[MRG + 2]`` on approval, making it eligible
+   for merging. An incomplete contribution -- where you expect to do more
+   work before receiving a full review -- should be prefixed ``[WIP]`` (to
```

Comment:
Should we move to using the draft PR feature of GitHub?

Reply:
I'd be happy to. Will ask the others